src/v2/guide/security.md

```diff
@@ -0,0 +1,166 @@
+
+For example, services like CodePen and JSFiddle allow user-provided content to be executed, but it's in a context where this is expected and sandboxed to some extent inside iframes. In the cases when an important feature inherently requires some level of vulnerability, it's up to your team to weigh the importance of the feature against the worst-case scenarios the vulnerability enables.
+
+### Injecting HTML
+
+As you learned earlier, Vue automatically escapes HTML content, preventing you from accidentally injecting executable HTML into your application. However, in cases where you know the HTML is safe, you can explicitly render HTML content:
+
+- Using a template:
+  ```html
+  <div v-html="userProvidedHtml"></div>
+  ```
+
+- Using a render function:
+  ```js
+  h('div', {
+    domProps: {
+      innerHTML: this.userProvidedHtml
+    }
+  })
+  ```
+
+- Using a render function with JSX:
+  ```jsx
+  <div domPropsInnerHTML={this.userProvidedHtml}></div>
+  ```
+
+<p class="tip">Note that user-provided HTML can never be considered 100% safe unless it's in a sandboxed iframe or in a part of the app where only the user who wrote that HTML can ever be exposed to it. Additionally, allowing users to write their own Vue templates brings similar dangers.</p>
+
+### Injecting URLs
+
+In a URL like this:
+
+```html
+<a v-bind:href="userProvidedUrl">
+  click me
+</a>
+```
+
+There's a potential security issue if the URL has not been "sanitized" to prevent JavaScript execution using `javascript:`. There are libraries such as [sanitize-url](https://www.npmjs.com/package/@braintree/sanitize-url) to help with this, but note:
+
+<p class="tip">If you're ever doing URL sanitization on the frontend, you already have a security issue. User-provided URLs should always be sanitized by your backend before even being saved to a database. Then the problem is avoided for _every_ client connecting to your API, including native mobile apps. Also note that even with sanitized URLs, Vue cannot help you guarantee that they lead to safe destinations.</p>
+
+### Injecting Styles
+
+Looking at this example:
+
+```html
+<a
+  v-bind:href="sanitizedUrl"
+  v-bind:style="userProvidedStyles"
+>
+  click me
+</a>
+```
+
+let's assume that `sanitizedUrl` has been sanitized, so that it's definitely a real URL and not JavaScript. With the `userProvidedStyles`, malicious users could still provide CSS to "click jack", e.g. styling the link into a transparent box over the "Log in" button. Then if `https://user-controlled-website.com/` is built to resemble the login page of your application, they might have just captured a user's real login information.
+
+You may be able to imagine how allowing user-provided content for a `<style>` element would create an greater vulnerability, giving that user full control over how to style the entire page:
+
+```html
+<style>{{ userProvidedStyles }}</style>
+```
```

Comment:
Should note here that Vue will refuse to render `<style>` tags inside templates.